widget/widget_date_input.go

```diff
@@ -0,0 +1,208 @@
+package widget
+
+import (
+	"fyne.io/fyne/v2"
+	"fyne.io/fyne/v2/canvas"
+	"fyne.io/fyne/v2/container"
+	"fyne.io/fyne/v2/theme"
+	"fyne.io/fyne/v2/widget"
+	"time"
+)
+
+var _ fyne.Focusable = (*JDateInputWidget)(nil)
+var _ fyne.Tappable = (*JDateInputWidget)(nil)
+
+// Custom Date Input Widget using RichText
+type JDateInputWidget struct {
+	widget.BaseWidget
+
+	// currently selected item data
+	currentDate time.Time
+
+	// current selected section
+	curSection string
+
+	// select section yes/no
+	selectSection int
+
+	// rich-text widget
+	richText *widget.RichText
+
+	// textSegments
+	daySeg   *widget.TextSegment
+	monthSeg *widget.TextSegment
+	yearSeg  *widget.TextSegment
+
+	// border
+	border *canvas.Rectangle
+
+	// layout
+	gl *fyne.Container
+
+	// app
+	fyneApp fyne.App
+
+	// main window
+	mainWindow fyne.Window
+}
+
+func (t *JDateInputWidget) addTime(secName string, v int) {
+	if secName == "d" {
+		t.currentDate = t.currentDate.AddDate(0, 0, v)
+	}
+
+	if secName == "m" {
+		t.currentDate = t.currentDate.AddDate(0, v, 0)
+	}
+
+	if secName == "y" {
+		t.currentDate = t.currentDate.AddDate(v, 0, 0)
+	}
+	t.updateDisplay()
+}
+
+func (t *JDateInputWidget) updateDisplay() {
+
+	t.daySeg.Text = t.currentDate.Format("02-")
+	t.monthSeg.Text = t.currentDate.Format("Jan-")
+	t.yearSeg.Text = t.currentDate.Format("2006")
+
+	t.daySeg.Style.TextStyle.Bold = false
+	t.monthSeg.Style.TextStyle.Bold = false
+	t.yearSeg.Style.TextStyle.Bold = false
+
+	if t.selectSection == 1 {
+		if t.curSection == "d" {
+			t.daySeg.Style.TextStyle.Bold = true
+		} else if t.curSection == "m" {
+			t.monthSeg.Style.TextStyle.Bold = true
+		} else if t.curSection == "y" {
+			t.yearSeg.Style.TextStyle.Bold = true
+		}
+	}
+	t.richText.Refresh()
+}
+
+// Start: implement focusable interface
+func (t *JDateInputWidget) FocusGained() {
+	t.border.FillColor = theme.FocusColor()
+	t.border.StrokeColor = theme.PrimaryColor()
+	t.updateDisplay()
+	t.Refresh()
+}
+
+func (t *JDateInputWidget) FocusLost() {
+	t.border.FillColor = theme.BackgroundColor()
+	t.border.StrokeColor = theme.InputBorderColor()
+	t.daySeg.Style.TextStyle.Bold = false
+	t.monthSeg.Style.TextStyle.Bold = false
+	t.yearSeg.Style.TextStyle.Bold = false
+	t.Refresh()
+}
+
+func (t *JDateInputWidget) TypedRune(k rune) {
+	// needed
```

Comment:
Not needed? I know it is needed for the interface but the comment makes that unclear. If you don't need this method for anything other than satisfying the interface, you can just remove the comment